### PR TITLE
Adds pagination property to show page number on bottom of page

### PR DESCRIPTION
**Pull request by kentsalcedo**
Adds pagination property to display page number on the bottom on the bottom of the page in the format (1 of 5), default to false. Styles pagination and arrows. Set Article property pagination={true} in order for the pages to show.

signed-off-by: Kent Salcedo <kentsalcedo@webmocha.com>

### CODE REVIEW

File: src/js/components/Article.js [hunk 1]
```diff
@@ -422,6 +424,19 @@ export default class Article extends Component {
     }
   }
 
+  _renderPagination () {
+    if (this.state.atBottom) {
```
Comment (alansouzati) [inline src/js/components/Article.js]: can we move this check to the render function ?

File: src/js/components/Article.js [hunk 2]
```diff
@@ -422,6 +424,19 @@ export default class Article extends Component {
...
+      const childCount = React.Children.count(this.props.children);
+      return (
+        <Box className="article__pagination" align="center">
```
Comment (alansouzati) [inline src/js/components/Article.js]: can we use CLASS_ROOT?

```
className={`${CLASS_ROOT}__pagination`}
```

File: src/js/components/Article.js [hunk 3]
```diff
@@ -422,6 +424,19 @@ export default class Article extends Component {
...
+          <Box align="center">
+            <Paragraph size="large">{this.state.selectedIndex+1} of {childCount}</Paragraph>
```
Comment (alansouzati) [inline src/js/components/Article.js]: minor suggestion: can we add a space in the operator?

this.state.selectedIndex + 1

File: src/js/components/Article.js [hunk 4]
```diff
@@ -486,6 +501,11 @@ export default class Article extends Component {
       classes.push(this.props.className);
     }
 
+    let pagination;
+    if (this.props.pagination) {
```
Comment (alansouzati) [inline src/js/components/Article.js]: Are we showing pagination for Article with direction column? I believe the pagination is only applicable for `direction="row"`

File: src/scss/grommet-core/_objects.article.scss [hunk 5]
```diff
@@ -59,8 +59,16 @@
   position: fixed;
   z-index: 10;
 
-  .button__icon {
-    padding: $inuit-base-spacing-unit;
```
Comment (alansouzati) [inline src/scss/grommet-core/_objects.article.scss]: why are we removing the padding for button__icon inside article?

**Comment by kentsalcedo** [comment]
@alansouzati - Hi Alan, I some issues after rebasing this PR so I submitted PR https://github.com/grommet/grommet/pull/565 which took into consideration the changes you suggested

**Pull request closed without merging**